source/server/overload_manager_impl.h

```diff
@@ -69,6 +70,11 @@ class OverloadManagerImpl : Logger::Loggable<Logger::Id::main>, public OverloadM
   // After this returns, overload manager clients should not receive any more callbacks
   // about overload state changes.
   void stop();
+  // Updates stats within specific resource monitor.
+  // Currently only integer values for stats are supported.
+  // todo(nezdolik) switch to ThreadLocalResourceStat struct
+  bool updateResourceStats(const std::thread::id thread_id, const std::string& resource,
```

Comment:
Where do you plan to call this method?  I wonder if we should consider an alternate implementation where the number of active downstream connections is tracked by some atomic stat counter and the resource update just computes the ratio between that counter and the configured max.

Reply:
I agree it would be good to step back and maybe do a short doc on how we plan on using this? I had always though that we need to have a type of resource monitor that is "instantaneous" and doesn't necessarily do delayed update. I might be missing some details on recent OM changes, but is this currently supported? My thinking here is that we want to have instantaneous pressure updates so we could still use delayed actions, but we could also do a circuit breaker like check on every new connection and make sure it's always accurate (modulo atomic convergange) as we accept new connections. So basically we would replace the runtime calls we do today with OM calls.

Reply:
Agree, starting with design doc might be a better idea. Have not worked with overload manager code before, so may be missing some in depth details.
Was thinking that it would be reasonable to keep specific atomic counters in related monitors (or whatever code related for that specific resource), so not to clutter overload manager code, as number of monitored resources would grow. Also my initial thinking was to call overload manager method `updateResourceStats` from connection manager (as connection manager has dependency on overload manager). That method in overload manager would have propagated new values to resource and transitively to their corresponding monitor 
(downstream connections monitor in that case). Lastly, downstream connections monitor would track counters per thread and then recompute atomic global counter upon `updateResourceUsage` is invoked. But sounds like you would like to have that responsibility (of tracking per thread and aggregating global counters) somewhere else but not in monitor.
Let me prepare design doc.